skipruntime-ts/ts/tests/tests.ts

```diff
@@ -717,6 +714,100 @@ tests.push({
   run: testAsyncLazyRun,
 });
 
+class MockExternal implements ExternalCall<number, string, TJSON> {
+  async call(key: number, timestamp: number) {
+    await timeout(1000 * key); // wait for `key` seconds before returning
+    return { payload: "mock_result(" + key + ")" };
+  }
+}
+
+class TestCheckExternalResult extends ValueMapper<number, number, string> {
+  constructor(private asyncLazy: AsyncLazyCollection<number, string, TJSON>) {
+    super();
+  }
+  mapValue(value: number, key: number): string {
+    const asyncRes = this.asyncLazy.getOne(key);
+    return asyncRes.status == "success"
+      ? "success: " + asyncRes.payload
+      : asyncRes.status == "failure"
+        ? "error: " + asyncRes.error
+        : "loading...";
+  }
+}
+tests.push({
+  name: "testExternalCall",
+  inputs: [schema("input", [integer("id", true, true), integer("value")])],
+  outputs: [schema("output", [integer("id", true, true), text("value")])],
+  tokens: { token_4s: 4000 },
+  init: (
+    skstore: SKStore,
+    input: TableCollection<[number, number]>,
+    output: TableCollection<[number, string]>,
+  ) => {
+    const external = skstore.externalCall(
+      "token_4s",
+      MockExternal,
+    ) as AsyncLazyCollection<number, string, TJSON>;
+    input
+      .map(TestFromIntInt)
+      .map(TestCheckExternalResult, external)
+      .mapTo(output, TestToOutput);
+  },
+
+  run: async (
+    input: Table<[number, number]>,
+    output: Table<[number, string]>,
+  ) => {
+    const success = (id: number) => {
+      return { id, value: `success: mock_result(${id})` };
+    };
+    const loading = (id: number) => {
+      return { id, value: "loading..." };
+    };
+
+    input.insert([
+      [1, 10],
+      [2, 20],
+      [3, 30],
+    ]);
+
+    await timeout(500);
+    // all loading at t = 0.5s
+    check("testExternalCall", output.select({}), [
+      loading(1),
+      loading(2),
+      loading(3),
+    ]);
+    await timeout(1000);
+    //id=1 succeeded at t = 1.5s
+    check("testExternalCall", output.select({}), [
+      success(1),
+      loading(2),
+      loading(3),
+    ]);
+    await timeout(1000);
+    //id=1,id=2 succeeded at t = 2.5s
+    check("testExternalCall", output.select({}), [
+      success(1),
+      success(2),
+      loading(3),
+    ]);
+    await timeout(1000);
+    //id=1,id=2,id=3 succeeded at t = 3.5s
+    check("testExternalCall", output.select({}), [
+      success(1),
+      success(2),
+      success(3),
+    ]);
+    await timeout(1000);
+    //all loading at t = 4.5s (because refresh triggered at t = 4s)
+    check("testExternalCall", output.select({}), [
+      loading(1),
+      loading(2),
+      loading(3),
```

Comment:
Do we really want `loading` after initialization?
Should we use the latest value instead?